test/assert_eventually_with_defaults_test.exs

```diff
@@ -5,6 +5,43 @@ defmodule AssertEventually.WithDefaultsTest do
 
   alias AssertEventually.Utils.MockOperation
 
+  describe "leaves properly assigned variables" do
+    test "without = operator" do
+      send(self(), {1, [2, 3], %{d: 4}})
+      eventually assert_received {a, [b, c], %{d: d}}
```

Comment:
Well, it doesn't make sense to use it considering `assert_receive` exists. That being said, I wanted this `eventually` be compatible with other 3rd party macros that are created to work with ExUnit. `assert_received`/`assert_receive` are unique in a way that those simulate "match" without `=` operator (those literally use provided `pattern` to generate left hand side of `=`). I'm still looking for better way of supporting those, as it would be way, waaay easier to just assume that I'm looking for variables that are on left side of each `=`  operator, but that would require me to "pre-compile" provided macros to native AST (macro-less) which I can't find how to do.

Reply:
oh I see now, damn that is an issue indeed.. good luck haha! If I find some time next week I'll try looking at this as well if you still haven't finished it, very clever what you did here anyway!

Reply:
What is happening right now is that I'm taking any variables and making a huge list of excludes like pinned variables or arguments of function calls. There is a huge chance that I missed something, but I guess this still covers more cases then current impl.

The other thing is that there is a solution that is trivial and covers 100% of cases as it doesn't care about a context of variable, but there is a catch - if you would ever use:
```
eventually assert some_var = 1
```
and then never used `some_var`, you wound **not** get an unused var warning. This is because the solution I'm talking about pulls all variables form the expression and reassigns them with `generated: true` metadata which suppresses this warning.